src/common/offline_transformations/src/pruning/shrink_weights.cpp

```diff
@@ -41,15 +42,13 @@ static bool is_static_reshape_op(std::shared_ptr<ov::Node> node) {
         return false;
 
     const auto& input_shape = input.get_shape();
-    const auto output_shape = output_shape_const_op->cast_vector<int64_t>();
+    const auto& output_shape = output_shape_const_op->cast_vector<int64_t>();
     // below casts are needed due to VC warning C4244, literals are not enough in this case
     const int64_t input_elems =
         std::accumulate(input_shape.begin(), input_shape.end(), static_cast<int64_t>(1), std::multiplies<int64_t>());
     const auto output_elems =
         std::accumulate(output_shape.begin(), output_shape.end(), static_cast<int64_t>(1), std::multiplies<int64_t>());
-    if (output_elems <= 0 || input_elems == output_elems)
-        return false;
-    return true;
+    return input_elems != output_elems;
```

Comment:
I guess if output_shape contains only one `-1` is also static case, we can calcualate the actual value based on static input shape and other values in the output_shape (if they are positive)

Reply:
if output_shape contains only one -1, this function will return true.